PayloadPkg/FirmwareUpdate/FirmwareUpdate.c

```diff
@@ -1130,10 +1130,12 @@ UpdateSystemFirmware (
   //
   // 2. Check firmware version.
   //
-  Status = VerifyFwVersion (ImageHdr, FwPolicy);
-  if (EFI_ERROR (Status)) {
-    DEBUG ((DEBUG_ERROR, " VerifyFwVersion failed with Status = 0x%x\n", Status));
-    return Status;
+  if (CompareGuid(&ImageHdr->UpdateImageTypeId, &gSblFWUpdateImageFileGuid)) {
+    Status = VerifyFwVersion(ImageHdr, FwPolicy);
+    if (EFI_ERROR (Status)) {
+      DEBUG ((DEBUG_ERROR, " VerifyFwVersion failed with Status = 0x%x\n", Status));
+      return Status;
+    }

```

Comment:
VerifyFwVersion () will decide how to check the version, check which components, etc.
So better to update VerifyFwVersion ()  for this change.

Reply:
Agree with Guo

Reply:
Also CompareGuid() returns INTN.  So you need to use if (CompareGuid() != 0)

Reply:
Done, updated code according to the comments